### PR TITLE
Revert #87462 "Fix PopupMenu doesn't respect its ScrollContainer's margins"

### LINKED ISSUE #93265: Right clicking in the editor to open properties of a certain element instantly selects what's under cursor.
### Tested versions

Not reproducible in v4.3.dev3.official [36e943b6b]
Reproducible in v4.3.dev4.official [df78c0636]
Reproducible in v4.3.dev5.official [c9c17d6ca]
Reproducible in v4.3.dev6.official [64520fe67]
Reproducible in v4.3.beta1.official [a4f2ea91a]

### System information

Godot v4.3.beta1 - Arch Linux #1 SMP PREEMPT_DYNAMIC Fri, 31 May 2024 15:14:45 +0000 - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1650 (nvidia; 550.78) - Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz (8 Threads)

### Issue description

When you right click, it instantly selects something even though you didn't want it to select anything. 
To fix this, all you really need is to check if the mouse moved substantially from when the right click button was clicked to when the right click button is released.

### Steps to reproduce

https://github.com/godotengine/godot/assets/69520693/1ea39699-4aeb-4591-baa6-3c093dc7cc14

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by radiantgurl**
- reverts #87462
- closes #93265
- reopens #87461

### CODE REVIEW

**Comment by RadiantUwU** [comment]
The ideal solution to be is not to revert the PR that attempts to fix a bug and introduces another one, but to have both of them fixed instead.

File: editor/editor_command_palette.cpp [hunk 1]
```diff
@@ -36,7 +36,6 @@
 #include "editor/gui/editor_toaster.h"
 #include "editor/themes/editor_scale.h"
 #include "scene/gui/control.h"
-#include "scene/gui/margin_container.h"
```
Comment (akien-mga) [inline editor/editor_command_palette.cpp:39]: I would keep those includes, so they're not just present by way of an implementation detail of PopupMenu, which may be changed back in the future.

Comment (akien-mga) [reply editor/editor_command_palette.cpp]: I meant all the `margin_container.h` includes which are being removed by this revert, also in other files.

**Comment by RadiantUwU** [comment]
I swear to god Windows.....
It failed to download the D3D12 SDK with an HTTP error 503..

**Comment by akien-mga** [comment]
So, I've been waiting to see if more investigation could be done on #93265 to understand the nature of the issue.

Reverting this visual change seems to only be addressing a symptom of the bug by changing slightly a position so that the bug isn't triggered. I'm not too keen on making such a change without understanding the root cause of the issue, as it means that issue may happen in other areas of the editor, or in user projects.

So far neither @bruvzg nor I managed to reproduce the issue on Linux, so it seems to affect only a subset of users. We'd need to identify which.

So I'd prefer not to merge this and instead go back to further debugging on #93265. It's probably too late to find a fix for 4.3-stable, but this is something we should hopefully be able to solve quickly after and backport to 4.3.1.

**Pull request closed without merging**